src/NHibernate.Test/NHSpecificTest/DtcFailures/DtcFailuresFixture.cs

```diff
@@ -17,247 +17,308 @@ namespace NHibernate.Test.NHSpecificTest.DtcFailures
 				config.AddResource(MappingsAssembly + "." + file, assembly);
 
-            // Copied from CreateSchema method, but we use our own config.
-            new SchemaExport(config).Create(false, true);
-        }
-
-        private void BeforeBindMapping(object sender, BindMappingEventArgs e)
-        {
-            HbmProperty prop = e.Mapping.RootClasses[0].Properties.OfType<HbmProperty>().Single(p => p.Name == "NotNullData");
-            prop.notnull = true;
-            prop.notnullSpecified = true;
-        }
+			// Copied from CreateSchema method, but we use our own config.
+			new SchemaExport(config).Create(false, true);
+		}
 
-		[Test]
-		public void WillNotCrashOnDtcPrepareFailure()
+		protected override void OnTearDown()
 		{
-			var tx = new TransactionScope();
-			using (ISession s = sessions.OpenSession())
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
 			{
-				s.Save(new Person {NotNullData = null});  // Cause a SQL not null constraint violation.
+				s.CreateQuery("delete from System.Object").ExecuteUpdate();
+				t.Commit();
 			}
+		}
 
-			new ForceEscalationToDistributedTx();
+		private void BeforeBindMapping(object sender, BindMappingEventArgs e)
+		{
+			var prop = e.Mapping.RootClasses[0].Properties.OfType<HbmProperty>().Single(p => p.Name == "NotNullData");
+			prop.notnull = true;
+			prop.notnullSpecified = true;
+		}
 
-			tx.Complete();
-			try
+		[Test]
+		public void SupportsEnlistingInDistributed()
+		{
+			using (new TransactionScope())
 			{
-				tx.Dispose();
-				Assert.Fail("Expected failure");
+				ForceEscalationToDistributedTx.Escalate();
+
+				Assert.AreNotEqual(
+					Guid.Empty,
+					System.Transactions.Transaction.Current.TransactionInformation.DistributedIdentifier,
+					"Transaction lacks a distributed identifier");
+
+				using (var s = sessions.OpenSession())
+				{
+					s.Save(new Person { CreatedAt = DateTime.Now });
+					// Ensure the connection is acquired (thus enlisted)
+					Assert.DoesNotThrow(s.Flush, "Failure enlisting a connection in a distributed transaction.");
+				}
 			}
-			catch (AssertionException)
+		}
+
+		[Test]
+		public void SupportsPromotingToDistributed()
+		{
+			using (new TransactionScope())
 			{
-				throw;
+				using (var s = sessions.OpenSession())
+				{
+					s.Save(new Person { CreatedAt = DateTime.Now });
+					// Ensure the connection is acquired (thus enlisted)
+					s.Flush();
+				}
+				Assert.DoesNotThrow(() => ForceEscalationToDistributedTx.Escalate(),
+					"Failure promoting the transaction to distributed while already having enlisted a connection.");
+				Assert.AreNotEqual(
+					Guid.Empty,
+					System.Transactions.Transaction.Current.TransactionInformation.DistributedIdentifier,
+					"Transaction lacks a distributed identifier");
 			}
-			catch (Exception) {}
 		}
 
 		[Test]
-		public void Can_roll_back_transaction()
+		public void WillNotCrashOnDtcPrepareFailure()
 		{
-			if (Dialect is FirebirdDialect)
```

Comment:
Does truly not support it (do not rollback from dtc, let changes in db), but now told by the dialect.